### DISCUSSION TITLE
htmlspecialchars(): Argument #1 ($string) must be of type string, Spatie\Html\Html given

### DISCUSSION THREAD

**Discussion by Ohene** [spatie/laravel-html#192] [Q&A]
Hello, I am new to Spatie/laravel-html and I have encountered this error. 
htmlspecialchars(): Argument #1 ($string) must be of type string, Spatie\Html\Html given

This happens when I call:
{{ html()->model($employee) }}
{{ html()->input('name') }}

Detail error display:
C:\xampp\htdocs\project\vendor\laravel\framework\src\Illuminate\Support\helpers.php : 124
htmlspecialchars

Can someone help know what I am doing wrong?

**Comment by fabricio-fn** [comment]
The model() method is normally used to create a form linked to a model, but it looks like you are trying to use it without providing a form name.

This may result in the creation of a Spatie\Html\Html object instead of a string.

try this

{!! html()->model($employee)->open() !!}
{{ html()->input('name') }}
{!! html()->closeModel() !!}}

**Reply by Ohene** [reply]
This is the error message I get when with this option:

"message": "Method Spatie\\Html\\Html::open does not exist.",
    "exception": "Spatie\\LaravelIgnition\\Exceptions\\ViewException",
    "file": "C:\\xampp\\htdocs\\holiday\\vendor\\laravel\\framework\\src\\Illuminate\\Macroable\\Traits\\Macroable.php",
    "line": 112,

Is $employee not supposed to be a model?

**Comment by imhugos** [comment]
I'm sure there must be a better way, but so far I'm able to get away with this:
`<?php html()->model($employee); ?>` (don't use double curly brackets)

**Comment by terremoth** [comment]
I am facing this problem too. The documentation seems confused about this.

**Comment by charlesbulger** [comment]
same here. laravel 11 forcing this. 

specifically i need a version of the laravelcollective setModel method to add the model to an already-opened form.

**Comment by terremoth** [comment]
The documentation seems to be wrong.

@charlesbulger @imhugos @Ohene 

You need to do, on blade: 

```blade
{{ html()->form(...)->open() }}
```

THEN

```blade
@php html()->model($yourModel); @endphp
```

This solves.

then continue with the inputs and `html()->form()->close()` .

The problem seems to occur when you use {{ }} or {!! !!}, this behind the scenes is an "echo/print" so it will try to print/echo the `html()->model($yourModel)` statement.

**Reply by charlesbulger** [reply]
Thanks! This will work. I ended up adding a blade directive in a Provider to clean it up a little:

`Blade::directive('setModel', function ($row) {
            return "<?php html()->model($row); ?>";
        });`
        
Then in view:

`@setModel($row ?? null)`

**Comment by raveren** [comment]
The current repository has the issue tracker disabled, it took me a fair bit of searching to find this thread :(

This is most definitely an issue, the example in the first page of documentation is dying with a critical Type error:
![image](https://github.com/spatie/laravel-html/assets/364220/d957c4c1-afc1-4dfa-88cc-61025602ab84)

I think it would make the most sense if the form would be accepting a model: `html()->form(/**/)->model($myModel)` but I'm wondering why it was not done like this in the first place. 

@freekmurze if you agree I could attempt a PR to add the method to Form, even though it's quite a big change.

I don't see another option without a BC break.

**Comment by terremoth** [comment]
@raveren 
> but I'm wondering why it was not done like this in the first place.

I asked myself the exactly same

**Comment by drelkomy** [comment]
htmlspecialchars(): Argument #1 ($string) must be of type string, array given      
     @forelse($cart_items as $item)
                                <tr wire:key='{{ $item['product_id'] }}'>
                                  <td class="py-4">
                                          <div class="flex items-center">
                                            <img class="h-16 w-16 mr-4" src="{{url('storage', $item['image'])}}" alt="{{$item['name']}}">
                                            <span class="font-semibold">{{$item['name']}}</span>
                                          </div>
                                        </td>
                                        <td class="py-4">{{Number::currency($item['unit_amount'], 'EGP') }}</td>
                                        <td class="py-4">
                                          <div class="flex items-center">
                                            <button class="border rounded-md py-2 px-4 mr-2">-</button>
                                            <span class="text-center w-8">{{$item['quantity']}}</span>
                                            <button class="border rounded-md py-2 px-4 ml-2">+</button>
                                          </div>
                                        </td>
                                        <td class="py-4">{{Number::currency($item['total_amount'], 'EGP') }}</td>
                                        <td>
                                            <button wire:click='removeItem({{$item['product_id']}})' class="bg-slate-300 border-2 border-slate-400 rounded-lg px-3 py-1 hover:bg-red-500 hover:text-white hover:border-red-700">Remove</button>
                                        </td>
                                      </tr>
                                @empty
                                <tr>
                                    <td colspan="5" class="text-center py-4 text-4xl font-semibold text-slate-500 ">
                                        No items Available in Cart!
                                    </td>
                                </tr>
                                @endforelse

**Comment by miken32** [comment]
For the record, this has worked fine for us for years:
```
{{ html()->model($instance)->form("put")->route("something.update", $instance->id)->open() }}

    {{ html()->label("Name", "name") }}
    {{ html()->text("name") }}

{{ html()->form()->close() }}
```
